### ISSUE TITLE
Using the ABC

### ISSUE THREAD

**Issue by sanskriti-s** [ecrl/ecabc#31]
As it currently stands, the code currently requires a lot of knowledge about ABCs from the user. We should make it so they have the choice of putting basic inputs like the cut off point, the fitness function etc and it runs.

**Comment by hgromer** [comment]
We moved away from this on purpose to make the code cleaner, and give more flexibility to the user. What do you think @tjkessler

**Comment by tjkessler** [comment]
A couple things:

- separating functions **does** give more flexibility/control to the user
    - as long as good documentation exists to explain each method in layman's terms, it should be okay
- _however_: if the execution order of the functions is not flexible (i.e. it's absolutely necessary that calc_average, calc_new_positions, check_positions are executed _in that order_), then giving the user this much control is useless - these functions could be nested inside a general "run for N iterations or until condition is met" function for the user

Ultimately we need to ask ourselves what the tangible benefits of having separate functions are.

**Comment by hgromer** [comment]
I think having both options would be beneficial. I don't want to go back to the 1.X standard of doing things, where we would terminate for the user. This produced horribly tangled, unnecessary code that will scale horribly. 

We could also have a 'run_iteration()' method, which would nest the functions and run them the way they do in the sample file. However I think exposing these individual methods would also be beneficial in giving the user more control.

**Comment by tjkessler** [comment]
I like the idea of bundling calc_average, calc_new_positions and check_positions into run_iteration, while keeping them public (single-underscore method notation, allowing public access but discouraging use unless the user knows what they are doing).

So, instead of:
```python
for _ in range(num_iterations):
    abc.calc_average()
    abc.calc_new_positions()
    abc.check_positions()
```
It could look like:
```python
for _ in range(num_iterations):
    abc.run_iteration()
```
This gives the user control over how many iterations to run for, and they only have to nest one call inside their loop.

And something similar for threshold termination:
```python
while abc.best_performer[0] > THRESHOLD:
    abc.run_iteration()
```
And if the user wants to work with the ABC in a lower level:
```python
for _ in range(num_iterations):
    abc._calc_average()
    abc._calc_new_positions()
    abc._check_positions()
```

**Comment by hgromer** [comment]
Agreed. This has now been updated for 2.1.0. Just a note. Because there are no do-while loops in python, the correct format is 

```python
while True:
    abc.run_iteration()
    if abc.best_performer[0] >/==/< THRESHOLD:
        break
```
